### PR TITLE
SMTP Fallback

### LINKED ISSUE #16: Fallback SMTP
Configurable options for Fallback

1. zimbraweb transport
2. SMTP with relay
3. Direct SMTP

Option 2 would need to be configured with an SMTP Server that will pass Mails for any Domain.

Option 3 should be disabled by default, but can be enabled by the User. It will also most likely fail, since residential IPs are blocked by most E-Mail providers.

### PR DESCRIPTION
**Pull request by JM-Lemmi**
Closes #16

Added Features
* configuration `smtp_fallback`: set to `enabled` will send the mail via smtp to a relay, if it is unparsable
* configuration `smtp_fallback_relay_host`: ip or hostname of relay
* milter now advises for smtp fallback in rejected mail
* sendmail automatically uses fallback if eml.is_parsable returns false

Todo for release:
- [x] publish python-zimbraweb v2.1
- [x] add to Readme
- [x] add docker-compose with reccomended paired smtp relay
- [x] remove /tests

Known limitations
* only plain smtp on port 25 for relay. No authentication

### CODE REVIEW

**Comment by JM-Lemmi** [comment]
This feature is now in a release-candidate state. So please go ahead with bugtesting it.

**Comment by JM-Lemmi** [comment]
Waiting on #44

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+services:
+  zimbraweb:
+    image: 'ghcr.io/cirosec-studis/zimbraweb-smtp-bridge:latest'
+    restart: unless-stopped
+    ports:
+    - '587:587'
+    environment:
+    - ENVCONFIG=true
+    - smtp_fallback=enabled
+    - smtp_fallback_relay_host=realyhost
```
Comment (frereit) [inline docker-compose.yml:13]: typo `realyhost` -> `relayhost`, right?

Comment (JM-Lemmi) [reply docker-compose.yml:13]: Yes of course.

**Review by frereit — APPROVED** [review]

**Pull request merged**